### PR TITLE
[youku] Fix list extraction.(close #15135)

### LINKED ISSUE #15135: [Youku] Crash when episode count is wrong
### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2017.12.31*. If it's not, read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2017.12.31**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [x] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---
Here is a test url: http://list.youku.com/show/id_z20eb4acaf5c211e3b2ad.html

```bash
youtube-dl http://list.youku.com/show/id_z20eb4acaf5c211e3b2ad.html\?spm\=a2h0j.8191423.module_basic_title.5\~A\!2 -v
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'http://list.youku.com/show/id_z20eb4acaf5c211e3b2ad.html?spm=a2h0j.8191423.module_basic_title.5~A!2', u'-v']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2017.12.23
[debug] Python version 2.7.13 - Darwin-17.2.0-x86_64-i386-64bit
[debug] exe versions: ffmpeg 3.3.2, ffprobe 3.3.2
[debug] Proxy map: {}
[youku:show] z20eb4acaf5c211e3b2ad: Downloading webpage
[youku:show] z20eb4acaf5c211e3b2ad: Downloading initial playlist data page
[youku:show] z20eb4acaf5c211e3b2ad: Downloading playlist data page 2
ERROR: An extractor error has occurred. (caused by KeyError(u'html',)); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 438, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/youku.py", line 292, in _real_extract
    'stage': reload_id,
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/youku.py", line 254, in _extract_entries
    transform_source=lambda s: js_to_json(strip_jsonp(s)))['html']
KeyError: u'html'
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 784, in extract_info
    ie_result = ie.extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 451, in extract
    raise ExtractorError('An extractor error has occurred.', cause=e)
ExtractorError: An extractor error has occurred. (caused by KeyError(u'html',)); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

The episode count is 16 however it has three modules 1-40 41-80 81-86.

Obviously it's youku's fault.

### PR DESCRIPTION
**Pull request by CoXier**
Change-Id: I2e9c920143f4f16012252625943a8f18b8ff40eb

## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### CODE REVIEW

**Comment by CoXier** [comment]
Hi, why does travis fail? 

From https://travis-ci.org/rg3/youtube-dl/jobs/324016795, I get log.

```bash
test_facebook_matching (test.test_all_urls.TestAllURLsMatching) ... ok
test_keywords (test.test_all_urls.TestAllURLsMatching) ... ok
test_no_duplicated_ie_names (test.test_all_urls.TestAllURLsMatching) ... ok
test_no_duplicates (test.test_all_urls.TestAllURLsMatching) ... 
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated
```
**So is test_no_duplicates not completed?**

But `TestAllURLsMatching` passed on my local pc.

```bash
python test/test_all_urls.py TestAllURLsMatching
.................
----------------------------------------------------------------------
Ran 17 tests in 13.497s

OK

```

**Comment by yan12125** [comment]
@CoXier: I've restarted the build.

**Comment by CoXier** [comment]
Thanks @yan12125 .

**Comment by CoXier** [comment]
@yan12125 Hi, does this pr have something to do?

File: youtube_dl/extractor/youku.py [hunk 1]
```diff
@@ -245,13 +245,20 @@ class YoukuShowIE(InfoExtractor):
...
     def _extract_entries(self, playlist_data_url, show_id, note, query):
         query['callback'] = 'cb'
-        playlist_data = self._download_json(
-            playlist_data_url, show_id, query=query, note=note,
-            transform_source=lambda s: js_to_json(strip_jsonp(s)))['html']
+        try:
+            playlist_data = self._download_json(
+                playlist_data_url, show_id, query=query, note=note,
+                transform_source=lambda s: js_to_json(strip_jsonp(s)))['html']
+        except KeyError:
```
Comment (yan12125) [inline youtube_dl/extractor/youku.py]: Is the KeyError come from ```['html']```? If so using ```.get('html')``` and check playlist_data is better.

**Comment by CoXier** [comment]
Thanks for your review. @yan12125 Please review again.

**Comment by yan12125** [comment]
Thanks for the fix!

**Pull request merged**